Index: sysvinit-2.87dsf/src/sulogin.c
===================================================================
--- sysvinit-2.87dsf.orig/src/sulogin.c	2010-01-18 09:41:41.012271775 +0100
+++ sysvinit-2.87dsf/src/sulogin.c	2010-01-18 09:44:01.902272010 +0100
@@ -40,7 +40,8 @@
 #define F_PASSWD	"/etc/passwd"
 #define F_SHADOW	"/etc/shadow"
 #define BINSH		"/bin/sh"
-#define STATICSH	"/bin/sash"
+#define STATICSH	"/bin/static-sh"
+#define STATICSH_SASH	"/bin/sash"
 
 char *Version = "@(#)sulogin 2.85-3 23-Apr-2003 miquels@cistron.nl";
 
@@ -373,6 +374,11 @@
 	setenv("SHELL", STATICSH, 1);
 	execl(STATICSH, STATICSH, NULL);
 	perror(STATICSH);
+
+	/* Fall back to sash */
+	setenv("SHELL", STATICSH_SASH, 1);
+	execl(STATICSH_SASH, STATICSH_SASH, NULL);
+	perror(STATICSH_SASH);
 }
 
 void usage(void)
